.github/workflows/ci.yaml

```diff
@@ -124,6 +124,8 @@ jobs:
         run: |
             sudo apt-get install libfmt-dev
             sudo apt-get install libmsgpack-dev
+            sudo apt-get update
```

Comment:
Do update before installing fit and msg pack too? Consider all installs on one line to reduce noise?

Reply:
Btw, why don't we install dependancies from source, like we do for wheels? Then we don't need all this branching with different install tools on different plattforms. A re-usable action would remove a lot of noise imo

Reply:
We actually have too many ways of installing dependencies. Here, in docker and in wheels.
I tried but didn't find a good way to unify it yet.
And as we can't unify it, I don't think we can create an action for it now, as everything is used differently.

Reply:
We can have an action, but only for the workflows. Although it doesn't unify anything, it would clean up these build scripts. Anyway, this PR didn't really introduce any of this, so I'm approving and we can maybe do something about this in the future